pkg/scalers/prometheus_scaler.go

```diff
@@ -297,6 +298,15 @@ func (s *prometheusScaler) ExecutePromQuery(ctx context.Context) (float64, error
 		}
 	}
 
+	if math.IsInf(v, 0) {
+		if s.metadata.ignoreNullValues {
+			return 0, nil
+		}
+		err := fmt.Errorf("promtheus query returns %f", v)
+		s.logger.Error(err, "Error recovering prometheus value")
```

Comment:
What does "recovering" mean here?
We know the value is infinite; I don't see any point stating more than that as the problem.

(I see that some errors log in this function and some do not; I was unable to detect the pattern as to which is appropriate for infinite values)

Reply:
I have updated the message because `Error converting prometheus value` is more real due the issue is with infinite values.

My only concern is if we should support to handle it as null values (returning 0) or it should be an error directly because this can produce huge peaks scaling out/in. 
One case with -infinite where scaling to zero is enabled, could produce an incorrect scale to zero but in the other hand, that could happen too if the query returns null values, that's why IDK if we should handle it in the same way as nulls